### ISSUE TITLE
Create a new Data Type based on any untyped FEEL or Boxed Expression on the DMN Editor

### ISSUE THREAD

**Issue by tiagobento** [apache/incubator-kie-tools#3759]
```[tasklist]
### Acceptance criteria
- [ ] Write a nice description for this issue.
- [ ] Add acceptance criteria here.
```

**Comment by jomarko** [comment]
Comment this to get notifications about this planned feature. Sounds interesting.

**Comment by baldimir** [comment]
I tried to work with enums today - it is not possible to create an Enum type constraint when creating a new data type (Enumeration is greyed out). I need to save the type first and edit again to convert it to enum. It would be nice to be 
able to create an enum right when creating the type. Mentioning it here so it could be considered as part of this ticket. 

![image (4)](https://github.com/apache/incubator-kie-tools/assets/125682/25956d8c-2bed-4760-bf75-045bab4c0dc6)

**Comment by jomarko** [comment]
@baldimir hi, the issue you reported above is an issues related to this PR https://github.com/kiegroup/kie-tools/pull/1323

It is because fresh added data type entry is `Any` by default. I see in you screenshot it is `string`, but my guess is you didn't save your type before you opened constraint dialogue, am I correct?

Still I agree this needs to be improved. However probably it is not part of apache/incubator-kie-tools#3759

**Comment by baldimir** [comment]
Yes, it is because I didn't save. That is the thing I tried to suggest. Sorry, if it was not clear. It should be able to define the type as enum right when someone is creating it. The type information is already there (as you noticed - I selected string before opening the dialog). One can select e.g. string as I did and then open the type constraints dialog.

**Comment by yesamer** [comment]
I suggested @baldimir report that case here, just to be sure the new component will support that scenario.
